lib/bigcommerce-oauth/connection.rb

```diff
@@ -43,6 +54,8 @@ def request(method, path, options, headers={})
           elsif response.code == 204
             nil
           end
+        rescue RestClient::TooManyRequests => e
```

Comment:
Why do you catch and raise?

Reply:
Otherwise it will be caught by the more generic `RestClient::RequestFailed` in line 63.